src/app/pages/sharing/nfs/nfs-list/nfs-list.component.ts

```diff
@@ -15,92 +22,179 @@ import { WebSocketService } from 'app/services/ws.service';
+                  },
+                });
+              },
+            });
+          },
+        },
+      ],
+    }),
+  ]);
+
+  isLoading$ = new BehaviorSubject<boolean>(true);
+  isNoData$ = new BehaviorSubject<boolean>(false);
+  hasError$ = new BehaviorSubject<boolean>(false);
+  emptyType$: Observable<EmptyType> = combineLatest([
+    this.isLoading$,
+    this.isNoData$,
+    this.hasError$,
+  ]).pipe(
+    switchMap(([isLoading, isNoData, isError]) => {
+      if (isLoading) {
+        return of(EmptyType.Loading);
+      }
+      if (isError) {
+        return of(EmptyType.Errors);
+      }
+      if (isNoData) {
+        return of(EmptyType.NoPageData);
+      }
+      return of(EmptyType.NoSearchResults);
+    }),
+  );
 
   constructor(
-    private slideInService: IxSlideInService,
-    protected ws: WebSocketService,
-    private errorHandler: ErrorHandlerService,
-    private dialog: DialogService,
+    private appLoader: AppLoaderService,
+    private ws: WebSocketService,
     private translate: TranslateService,
+    private dialog: DialogService,
+    private errorHandler: ErrorHandlerService,
+    private slideInService: IxSlideInService,
+    private cdr: ChangeDetectorRef,
+    protected emptyService: EmptyService,
   ) {}
 
-  afterInit(entityList: EntityTableComponent<NfsShare>): void {
-    this.entityList = entityList;
+  ngOnInit(): void {
+    this.loadData();
   }
 
-  confirmDeleteDialog = {
-    message: shared.delete_share_message,
-    isMessageComplete: true,
-    button: this.translate.instant('Unshare'),
-    buildTitle: (share: NfsShare) => `${this.translate.instant('Unshare')} ${share.path}`,
-  };
-
-  doAdd(): void {
-    const slideInRef = this.slideInService.open(NfsFormComponent);
-    slideInRef.slideInClosed$.pipe(filter(Boolean), untilDestroyed(this)).subscribe(() => this.entityList.getData());
+  loadData(): void {
+    this.isLoading$.next(true);
+    this.ws.call('sharing.nfs.query').pipe(
+      untilDestroyed(this),
+    ).subscribe({
+      next: (shares: NfsShare[]) => {
+        this.nfsShares = shares;
+        this.dataProvider.setRows(this.nfsShares);
+        this.isLoading$.next(false);
+        this.isNoData$.next(!this.nfsShares?.length);
+        this.setDefaultSort();
+        this.cdr.markForCheck();
+      },
+      error: () => {
+        this.dataProvider.setRows([]);
+        this.isLoading$.next(false);
+        this.hasError$.next(true);
+        this.cdr.markForCheck();
+      },
+    });
   }
 
-  doEdit(id: number): void {
-    const nfsShare = this.entityList.rows.find((row) => row.id === id);
-    const slideInRef = this.slideInService.open(NfsFormComponent, { data: nfsShare });
-    slideInRef.slideInClosed$.pipe(filter(Boolean), untilDestroyed(this)).subscribe(() => this.entityList.getData());
+  setDefaultSort(): void {
+    this.dataProvider.setSorting({
+      active: 1,
+      direction: SortDirection.Asc,
+      propertyName: 'id',
```

Comment:
In webui it's show that `Description` is being sorted. This is not ideal, because description can be empty, best to sort by path.